poetry/core/masonry/builders/builder.py

```diff
@@ -181,9 +184,7 @@ def find_files_to_add(
                     path=file, project_root=self._path, source_root=source_root
                 )
 
-                if self.is_excluded(
-                    include_file.relative_to_project_root()
-                ) and isinstance(include, PackageInclude):
+                if self.is_excluded(file.relative_to(source_root)):
```

Comment:
This changes include/exclude behavior, breaking three tests. Currently, include always trumps exclude (see also python-poetry/poetry#1336), which is too limiting for some use cases.

One solution, without introducing order-dependent include/exclude rules as in a a _MANIFEST.in_, is to give globbed paths a lower priority than fully-qualified paths. I feel this offers more flexibility while being intuitive at the same time. This would work for the examples given in python-poetry/poetry#1336, such as:
```ini
exclude = [
    "assets/sample/**/*"
]
include = [
    "assets/sample/subdir/one.yaml",
    "assets/sample/subdir/two.yaml"
]
```
But the following would now also work:
```ini
include = [
    "**/*"
]
exclude = [
    ".travis.yml"
]
```